### PR TITLE
feat: implement credential

### PR DESCRIPTION
**Pull request by dspo**
### Description

This PR add a resource named `credential` and implement Admin APIs for it.
Credential is used to hold the authentication credentials for the Consumer.
Credentials are used when multiple credentials need to be configured for a Consumer.

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Review by nic-6443 — APPROVED** [review]

**Comment by nic-6443** [comment]
@kayx23 Please help review the documentation part of this PR.

File: docs/en/latest/admin-api.md [hunk 1]
```diff
@@ -868,6 +868,71 @@ Since `v2.2`, we can bind multiple authentication plugins to the same consumer.
...
+        "key": "auth-one"
+      }
+    },
+    "desc": "hello world"
+}
+```
+
+### Example API usage
+
+Prerequisite: Consumer has been created.
```
Comment (kayx23) [inline docs/en/latest/admin-api.md:912]:
```suggestion
Prerequisite: Consumer `jack` has been created.
```

File: docs/en/latest/admin-api.md [hunk 2]
```diff
@@ -868,6 +868,71 @@ Since `v2.2`, we can bind multiple authentication plugins to the same consumer.
...
+
+Create Credential and specify the authentication plugin `key-auth`:
```
Comment (kayx23) [inline docs/en/latest/admin-api.md:914]: How about

```suggestion
Create the `key-auth` Credential for consumer `jack`:
```

File: docs/en/latest/terminology/credential.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+    ```shell
+    curl http://127.0.0.1:9180/apisix/admin/consumers \
+    -H "X-API-KEY: $admin_key" -X PUT -d '
+    {
+        "username": "jack"
+    }'
+    ```
+
+2. Create 2 `key-auth` 的 Credentials for the Consumer.
```
Comment (kayx23) [inline docs/en/latest/terminology/credential.md:79]: image

Mix of Chinese and English throughout this doc.

**Comment by moonming** [comment]
Is this a break change for using the authentication plugin on consumers?

**Comment by dspo** [comment]
> Is this a break change for using the authentication plugin on consumers?

It's not a break change. Users can still use Consumer in the same way as before, and the one change to Consumer is that it is no longer required to have at least one auth plugin.

File: docs/en/latest/admin-api.md (hunk 1 above)
Comment (dspo) [reply docs/en/latest/admin-api.md]: I resolved it.

File: docs/en/latest/admin-api.md (hunk 2 above)
Comment (dspo) [reply docs/en/latest/admin-api.md]: I resolved it.

**Review by moonming — APPROVED** [review]

**Comment by kayx23** [comment]
> @kayx23 Please help review the documentation part of this PR.

No additional comments from me for docs.

**Review by kayx23 — APPROVED** [review]
doc lgtm

**Review by nic-6443 — APPROVED** [review]

**Pull request merged**